zipline/utils/api_utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Quantopian, Inc.
```

Comment:
I considered placing the new code in `utils/api_support.py`, but @ssanderson felt that the logic in there was quite distinct from that which is now in `api_utils.py`. The logic in `api_utils.py` relates specifically to user input, while that in `api_support.py` are for internal control and checks.

`api_utils` would work with non-api-methods, but the methods in there have been specifically designed to decorate our api methods.

Perhaps `api_validation.py` would be a better name for `api_utils.py`?